### PR TITLE
Enable subsecond block times using unix nano for Time in BaseHeader

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview

Closes: #1147

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: block/manager.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func NewManager(
 		// channels are buffered to avoid blocking on input/output operations, buffer sizes are arbitrary
 		HeaderCh:          make(chan *types.SignedHeader, channelLength),
 		BlockCh:           make(chan *types.Block, channelLength),
-		blockInCh:         make(chan newBlockEvent, channelLength),
+		blockInCh:         make(chan newBlockEvent, 100000),
```
Comment (MSevey) [inline block/manager.go:180]: new constant please with justification.

**Review by MSevey — CHANGES_REQUESTED** [review]
minor change, otherwise looks good.

File: config/defaults.go [hunk 2]
```diff
@@ -21,7 +19,6 @@ var DefaultNodeConfig = NodeConfig{
 	Aggregator:     false,
 	LazyAggregator: false,
 	BlockManagerConfig: BlockManagerConfig{
-		BlockTime:   30 * time.Second,
```
Comment (gupadhyaya) [inline config/defaults.go:24]: i would not remove this. set the default here to 1s and remove the re-init in manager.go.

Comment (gupadhyaya) [reply config/defaults.go:24]: you can do the same for `DABlockTime`

Comment (Manav-Aggarwal) [reply config/defaults.go:24]: Done

File: config/defaults.go [hunk 3]
```diff
@@ -21,7 +21,8 @@ var DefaultNodeConfig = NodeConfig{
...
+		BlockTime:   1 * time.Second,
```
Comment (gupadhyaya) [inline config/defaults.go:24]: do you still need `defaultBlockTime` and `defaultDABlockTime` in the manager.go? and this code is probably not needed:
```
// defaultDABlockTime is used only if DABlockTime is not configured for manager
const defaultDABlockTime = 30 * time.Second

// defaultBlockTime is used only if BlockTime is not configured for manager
const defaultBlockTime = 1 * time.Second
...
if conf.DABlockTime == 0 {
		logger.Info("Using default DA block time", "DABlockTime", defaultDABlockTime)
		conf.DABlockTime = defaultDABlockTime
	}

	if conf.BlockTime == 0 {
		logger.Info("Using default block time", "BlockTime", defaultBlockTime)
		conf.BlockTime = defaultBlockTime
	}
```

Comment (Manav-Aggarwal) [reply config/defaults.go:24]: removing it breaks `rpc/json` tests. So it's needed.

**Comment by Manav-Aggarwal** [comment]
> @Manav-Aggarwal did you test with an aggregator and full node and ensured that blocks are synced by the full node? just ensuring that this change does not lead to any regressions.

yes

**Review by gupadhyaya — APPROVED** [review]

**Review by nashqueue — APPROVED** [review]

**Review by MSevey — APPROVED** [review]

**Pull request merged**